File: backend/app/schemas/audit.py
"""审计日志响应模型。"""
from datetime import datetime

from pydantic import BaseModel, Field


class AuditLogRead(BaseModel):
    """审计日志只读结构。"""

    id: int
    actor_id: int
    action: str = Field(..., description="动作标识")
    target_type: str = Field(..., description="目标类型")
    target_id: str = Field(..., description="目标标识")
    note: str | None = Field(None, description="操作备注")
    ip: str | None = Field(None, description="发起操作的客户端 IP")
    created_at: datetime

    class Config:
        orm_mode = True

File: frontend/src/api/audit.ts
import { httpClient } from "./http";

export interface AuditLogItem {
  id: number;
  actor_id: number;
  action: string;
  target_type: string;
  target_id: string;
  note: string | null;
  ip: string | null;
  created_at: string;
}

export async function fetchAuditLogs(limit = 20): Promise<AuditLogItem[]> {
  const { data } = await httpClient.get<{ data: AuditLogItem[] }>("/audits", {
    params: { limit }
  });
  return data.data;
}
